harvester_e2e_tests/integration/test_vm_functions.py

```diff
@@ -296,18 +296,19 @@ def test_softreboot(self, api_client, unique_vm_name, wait_timeout):
         assert "AgentConnected" == old_agent['type'], (code, data)
 
         api_client.vms.softreboot(unique_vm_name)
+        # Wait until agent disconnected (leaving OS)
         endtime = datetime.now() + timedelta(seconds=wait_timeout)
         while endtime > datetime.now():
             code, data = api_client.vms.get_status(unique_vm_name)
             if "AgentConnected" not in data['status']['conditions'][-1]['type']:
                 break
             sleep(5)
-
+        # then wait agent connected again (Entering OS)
         endtime = datetime.now() + timedelta(seconds=wait_timeout)
         while endtime > datetime.now():
             code, data = api_client.vms.get_status(unique_vm_name)
-            phase, conds = data['status']['phase'], data['status']['conditions']
-            if "Running" == phase and "AgentConnected" == conds[-1]['type']:
+            phase, conds = data['status']['phase'], data['status'].get('conditions', [{}])
+            if "Running" == phase and "AgentConnected" == conds[-1].get('type'):
```

Comment:
We may want to add a safeguard, something like:
```python
sample = [{}]
assert "AgentConnected" == sample[-1].get('type')
# creates:
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AssertionError
```
Which might be fine.
Is there a chance that `conditions` might not yet exist- but `phase` does - but because we set to a default `list(dict())` that the `AssertionError` would stop the test prematurely on the `assert "AgentConneccted" == sample[-1].get('type')`?

Reply:
nevermind :sweat_smile: